src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx

```diff
@@ -0,0 +1,243 @@
+              </Text>
+
+              <Text variant="md" fontWeight="400">
+                {stateLabel}
+              </Text>
+
+              {!!submission.actionLabel && (
+                <Text variant="md" fontWeight="400" color="orange100">
+                  {submission.actionLabel}
+                </Text>
+              )}
+
+              <Spacer y={1} />
+
+              <Text variant="sm" color="black60">
+                {stateHelpMessage}
+              </Text>
+
+              <Spacer y={2} />
+
+              {!!buttonURL && (
+                <Button
+                  variant={buttonVariant}
+                  // @ts-ignore
+                  as={RouterLink}
+                  onClick={() => {
+                    // TODO: Tracking
+                  }}
+                  to={buttonURL}
+                  width="100%"
+                  mb={2}
+                >
+                  {buttonLabel}
+                </Button>
+              )}
+            </Box>
+          </ModalDialog>
+        )}
+      </Media>
+
+      <Media greaterThanOrEqual="sm">
+        {!!submission.actionLabel && (
+          <Text variant="md" fontWeight="400" color="orange100">
+            {submission.actionLabel}
+          </Text>
+        )}
+
+        <Spacer y={1} />
+
+        <Text variant="sm" color="black60">
+          {stateHelpMessage}
+        </Text>
+
+        <Spacer y={2} />
+
+        {!!buttonURL && (
+          <Button
+            variant={buttonVariant}
+            // @ts-ignore
+            as={RouterLink}
+            onClick={() => {
+              // TODO: Tracking
+            }}
+            to={buttonURL}
+          >
+            {buttonLabel}
+          </Button>
+        )}
+      </Media>
+    </Box>
+  )
+}
+
+const submissionStatusFragment = graphql`
+  fragment MyCollectionArtworkSWASubmissionStatus_artwork on Artwork {
+    listedArtworksConnection(first: 1) {
+      edges {
+        node {
+          internalID
+        }
+      }
+    }
+    consignmentSubmission {
+      actionLabel
+      buttonLabel
+      internalID
+      state
+      stateLabel
+      stateHelpMessage
+    }
+  }
+`
+
+const getButtonURL = (
+  artwork: MyCollectionArtworkSWASubmissionStatus_artwork$data
+): string | null => {
+  const submission = artwork.consignmentSubmission
+
+  const listedArtwork = extractNodes(artwork.listedArtworksConnection)[0]
+
+  // The artwork has been listed on Artsy.
+  if (listedArtwork) {
+    return `/artwork/${listedArtwork?.internalID}`
+  }
+
+  if (!submission) return null
+
+  if (
+    ["DRAFT", "SUBMITTED", "RESUBMITTED", "PUBLISHED"].includes(
+      submission.state
+    )
+  ) {
+    return `/sell/submissions/${submission.internalID}/${INITIAL_STEP}`
+  }
+
+  if (["APPROVED"].includes(submission.state)) {
+    return `/sell/submissions/${submission.internalID}/${INITIAL_POST_APPROVAL_STEP}`
+  }
+
+  return null
+}
+
+const getStateHelpMessage = (submission, isListed): JSX.Element => {
```

Comment:
What do you think about moving `getStateHelpMessage` and `getButtonURL` to utils?